client/my-sites/domains/domain-search/index.jsx

```diff
@@ -208,6 +226,12 @@ class DomainSearch extends Component {
 		const { domainRegistrationMaintenanceEndTime } = this.state;
 
 		const hasPlanInCart = hasPlan( cart );
+		const hrefForDecideLater = addQueryArgs(
+			{
+				domainAndPlanPackage: true,
+			},
+			`/plans/yearly/${ selectedSite.domain }`
```

Comment:
Note: theoretically, using the site slug is a better idea than using the domain. The rationale: IIRC `siteSlug` is better suited for working with sites installed in subdirectories.

Reply:
Nice, [fixed here](https://github.com/Automattic/wp-calypso/pull/73017/commits/504282123d8ac50cb8978af6f669603f23727aba).